### PR TITLE
CMR-7512 Add s3-links to umm-json responses when available

### PR DESCRIPTION
**Pull request by jaybarra**
umm-json responses are validated against the schema in tests, extraneous keys were being rejected with the existing model therefore an update to the umm-c response was required.

The s3-links are not returned as part of the response where there are none, following existing behavior for the meta portion of a response item.

### CODE REVIEW

**Review by daniel-zamora — APPROVED** [review]

File: search-app/src/cmr/search/results_handlers/umm_json_results_handler.clj [hunk 1]
```diff
@@ -2,6 +2,8 @@
   "Handles Collection umm-json results format and related functions"
   (:require
    [cheshire.core :as json]
+   [clojure.edn :as edn]
+   [clojure.string :as str]
```
Comment (eereiter) [inline search-app/src/cmr/search/results_handlers/umm_json_results_handler.clj:6]: Delete both the edn and str namespace aliases, as they are not used in the file.

File: search-app/src/cmr/search/results_handlers/umm_json_results_handler.clj [hunk 2]
```diff
@@ -36,7 +38,9 @@
 
 (defmethod elastic-search-index/concept-type+result-format->fields [:collection :umm-json-results]
   [concept-type query]
-  results-helper/meta-fields)
+  (concat
+   results-helper/meta-fields
+   ["s3-bucket-and-object-prefix-names"]))
```
Comment (eereiter) [inline search-app/src/cmr/search/results_handlers/umm_json_results_handler.clj:43]: Why is this ["s3-bucket-and-object-prefix-names"] hardcoded here and not included with the other fields that are defined in the meta-fields list?  Please put s3-bucket-and-object-prefix-names into the meta-fields list.  The documentation in the meta-fields function states: "Defines the fields in elastic search we retrieve to populate the meta fields in the response." and s3-bucket-and-object-prefix-names is a field in elastic search we retrieve and it is being put into the meta fields response section.

File: umm-spec-lib/resources/json-schemas/collection/umm/v1.16.5/umm-search-results-json-schema.json [hunk 3]
```diff
@@ -0,0 +1,164 @@
+{
```
Comment (eereiter) [inline umm-spec-lib/resources/json-schemas/collection/umm/v1.16.5/umm-search-results-json-schema.json:1]: Next time when you update the schemas, please first copy the previous version to the new version and then add and commit then new directory before you make any changes.  This allows reviewers to be able to easily and quickly see the changes you have made to the schemas.

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj [hunk 4]
```diff
@@ -453,5 +458,10 @@
   [context collection & _]
   ;; Remove the related urls that sub-types that were not valid in the lower versions.
   (-> collection
-        (update :RelatedUrls remove-1-16-4-urls)
-        util/remove-nils-empty-maps-seqs))
+      (update :RelatedUrls remove-1-16-4-urls)
+      util/remove-nils-empty-maps-seqs))
```
Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj:462]: + for updating the spacing. Thank you.

**Review by eereiter — CHANGES_REQUESTED** [review]

File: umm-spec-lib/resources/json-schemas/collection/umm/v1.16.5/umm-search-results-json-schema.json (hunk 3 above)
Comment (jceaser) [reply umm-spec-lib/resources/json-schemas/collection/umm/v1.16.5/umm-search-results-json-schema.json:1]: can't this still be done either with a fresh branch or by rewriting history in git?

**Review by jceaser — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

**Pull request merged**